code/ai/aiturret.cpp

```diff
@@ -1904,7 +1904,9 @@ bool turret_fire_weapon(int weapon_num, ship_subsys *turret, int parent_objnum,
 							points = 1;
 						}
 
-						if (swp->primary_bank_ammo[bank_to_fire] >= 0) {
+						if (swp->primary_bank_ammo[bank_to_fire] >= points) {
+							swp->primary_bank_ammo[bank_to_fire] -= points;
+						} else if ((swp->primary_bank_ammo[bank_to_fire] >= 0) && !(The_mission.ai_profile->flags[AI::Profile_Flags::Prevent_negative_turret_ammo])) {
```

Comment:
This doesn't seem to make sense, actually. The ammo can still end up being negative. This would fire from each primary point, right? So isn't it still firing too many shots?  Looking for clarification.

Reply:
For the check on line 1909? Or on 1907?

Reply:
1909

Reply:
Correct, 1909 is just the default behavior, which allowed the ammo to go negative. :)

Reply:
And that's why you don't comment on PRs after a three day drive.

Reply:
Perhaps would have made more sense to put that line first (to correspond to where the old check was, and to add a comment about why). Be happy to add that if folks want.